File: easy/1408_string_matching_in_an_array/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  std::vector<std::string> stringMatching(std::vector<std::string> &words) {
    std::vector<std::string> substrings;

    for (size_t i = 0; i < words.size(); ++i) {
      for (size_t j = 0; j < words.size(); ++j) {
        if (i == j)
          continue;

        if (words[j].find(words[i]) != std::string::npos) {
          substrings.push_back(words[i]);
          break;
        }
      }
    }

    return substrings;
  }
};

File: easy/1408_string_matching_in_an_array/solution.py
from typing import List


class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        substrings = set()

        for i in range(len(words)):
            for j in range(len(words)):
                if i == j:
                    continue

                if words[j] in words[i]:
                    substrings.add(words[j])

        return list(substrings)
